lib/datadog/di/component.rb

```diff
@@ -53,12 +57,12 @@ def environment_supported?(settings)
           # TODO add tests?
           unless settings.dynamic_instrumentation.internal.development
             if Datadog::Core::Environment::Execution.development?
-              Datadog.logger.debug("Not enabling dynamic instrumentation because we are in development environment")
+              Datadog.logger.warn("Not enabling dynamic instrumentation because we are in development environment")
```

Comment:
Minor: Since this is a customer-visible string, I'd suggest not using the "we" since it's a bit confusing. Here's a potential alternative

```suggestion
              Datadog.logger.warn("Development environment detected; not enabling dynamic instrumentation")
```